src/routing/mod.rs

```diff
@@ -258,8 +258,14 @@ impl Routing {
         src: SrcLocation,
         dst: DstLocation,
         content: Bytes,
+        join_flag: bool,
```

Comment:
Currently binding it to send_message just to ensure the vault status will be periodically checked, so that the flag can be updated in time.

If a separate setter is neater, I will make the change.

Reply:
I think a separate setter is cleaner. If sn-node wants to set it always when they send a message, they can still do it, but I think they will want to only set it when they detect change in the section storage capacity or things like that. Bundling it together with message sending seems confusing and not that useful.